lib/galaxy/app.py

```
@@ -668,6 +668,10 @@ def __init__(self, configure_logging=True, use_converters=True, use_display_appl
         self._register_singleton(Registry, self.datatypes_registry)
         galaxy.model.set_datatypes_registry(self.datatypes_registry)
         self.configure_sentry_client()
+        # Load dbkey / genome build manager
+        self._configure_genome_builds(data_table_name="__dbkeys__", load_old_style=True)
+        # Tool Data Tables
+        self._configure_tool_data_tables(from_shed_config=False)
```

This takes a long time on main, but I guess there's no way around it ?
I had intentionally excluded it so that celery workers can cycle quickly.

---

I guess a follow-up optimization could be to make the data tables lazy so that the celery workers that don't need them can start up fast

---

We could also use a new apps abstraction - a bit heavier than this one - like we've done in other places. I really don't want this to slow down all the existing tasks either way so this is a good catch.

---

Ugh - these things are annotated at the top of that stack (most basic app protocols) so this is a bug fix either way. Maybe unwinding this for the other tasks would be too much surgery and we should just make the attributes lazy? I feel like making each table lazy would mask a lot of errors in the logs right? The main app should materialize all of this during launch so it doesn't start if there are errors - or do we swallow them all anyway?

---

https://github.com/galaxyproject/galaxy/compare/dev...jmchilton:galaxy:lazy_data_tables?expand=1

---

It's all swallowed, your branch looks good to me ?

---

Opened with #21163.